### PR TITLE
Optimization rolling update pd

### PR DESCRIPTION
**Pull request by liubo0127**
Divide the pd node into two groups(leader and follower), upgrade the follower first, and finally upgrade the leader.
@LinuxGit @superlzs0476 PTAL

### CODE REVIEW

File: rolling_update.yml [hunk 1]
```diff
@@ -54,8 +54,51 @@
...
+        status_code: 200
+      register: pd_status
+    - name: Failed when one node of pd is unhealthy
+      fail:
+        msg: "Some pd node is unhealth"
+      when: '"false" in pd_status.content'
+
+- hosts: pd_servers
+  any_errors_fatal: true
+  serial: 1
```
Comment (liubo0127) [inline rolling_update.yml]:
```suggestion
```

File: rolling_update.yml [hunk 2]
```diff
@@ -54,8 +54,58 @@
...
+        msg: "Some pd node is unhealth"
+      when: "'false' in pd_status.content"
+
+- hosts: pd_servers
+  any_errors_fatal: true
+  serial: 1
+  tags:
+    - pd
+  tasks:
+    - name: Get pd leader
```
Comment (LinuxGit) [inline rolling_update.yml]: reuse https://github.com/pingcap/tidb-ansible/blob/master/common_tasks/get_pd_leader.yml

File: rolling_update.yml [hunk 3]
```diff
@@ -54,8 +54,58 @@
...
+    - name: Check pd cluster status
+      uri:
+        url: "http://{{ ansible_host }}:{{ pd_client_port }}/pd/health"
+        method: GET
+        return_content: yes
+        status_code: 200
+      register: pd_status
+    - name: Failed when one node of pd is unhealthy
+      fail:
+        msg: "Some pd node is unhealth"
```
Comment (LinuxGit) [inline rolling_update.yml]: syntax error

File: excessive_rolling_update.yml [hunk 4]
```diff
@@ -54,8 +54,58 @@
...
+      register: pd_leader_info
+    - set_fact:
+        pd_leader: "{{ pd_leader_info.json | json_query('client_urls') }}"
+    - name: Set pd follower list
+      add_host:
+        name: "{{ ansible_host }}"
+        groups: pd_servers_followers
+        pd_client_port: "{{ pd_client_port }}"
+        deploy_dir: "{{ deploy_dir }}"
+        pd_peer_port: "{{ pd_peer_port }}"
```
Comment (LinuxGit) [inline excessive_rolling_update.yml]: It seems that pd_servers_followers groups don't have group variables in  
https://github.com/pingcap/tidb-ansible/blob/master/group_vars/pd_servers.yml and other customized  host variables.

File: rolling_update.yml [hunk 5]
```diff
@@ -54,8 +54,101 @@
         - current_version.stdout_lines[0].replace(' ','').split(':')[1] < "v2.0.1"
         - tidb_version >= "v2.1.0" or tidb_version == "latest"
 
+- hosts: pd_servers
+  any_errors_fatal: true
+  serial: 1
```
Comment (LinuxGit) [inline rolling_update.yml]: don't need `serial: 1`?

File: rolling_update.yml [hunk 6]
```diff
@@ -54,8 +54,101 @@
...
+    - name: Failed when one node of pd is unhealthy when enable_tls
+      fail:
+        msg: "Some pd nodes are unhealthy"
+      when:
+        - enable_tls|default(false)
+        - "'false' in pd_status_tls.content"
+
+- hosts: pd_servers
+  any_errors_fatal: true
+  serial: 1
```
Comment (LinuxGit) [inline rolling_update.yml]: ditto

**Pull request merged**